### PR TITLE
Feature/add move class

### PR DESCRIPTION
**Pull request by dktrotti**
Added a move class, damage is now dealt using EventManager/HitEvent

### CODE REVIEW

File: ShanghaiBloodSports/Assets/Scripts/AnimationController.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (animator != null)
+        {
+            switch (character.CurrentState)
+            {
+                case Character.State.NEUTRAL:
+                    animator.SetBool("back_walk", false);
```
Comment (Loonman) [inline ShanghaiBloodSports/Assets/Scripts/AnimationController.cs]: Oh unity, why are you so stringly typed

File: ShanghaiBloodSports/Assets/Scripts/AutoRotator.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(Character))]
```
Comment (Loonman) [inline ShanghaiBloodSports/Assets/Scripts/AutoRotator.cs]: 👍

File: ShanghaiBloodSports/Assets/Scripts/Character.cs [hunk 3]
```diff
@@ -1,52 +1,135 @@
...
+
+    private Move currentMove;
+    public Move CurrentMove {
+        get {
+            return currentMove;
+        }
         set {
-            health = value;
-            healthIndicator.text = $"Health: {health}\nGuard: {guard}";
+            if (currentMove == null ^ value == null)
```
Comment (Loonman) [inline ShanghaiBloodSports/Assets/Scripts/Character.cs]: Please add a comment to clarify

File: ShanghaiBloodSports/Assets/Scripts/Character.cs [hunk 4]
```diff
@@ -1,52 +1,135 @@
...
+        public string Path { get; private set; }
 
-        var relativeXPosition = opponent.transform.position.x - transform.position.x;
+        private HitboxPath(string path)
+        {
+            Path = path;
+        }
 
-        return relativeXPosition > 0;
+        public static HitboxPath UPPER_BODY = new HitboxPath("AnimationControl/Hitboxes/UpperBody");
```
Comment (Loonman) [inline ShanghaiBloodSports/Assets/Scripts/Character.cs]: 🤑

File: ShanghaiBloodSports/Assets/Scripts/MockInputBuffer.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections;
```
Comment (Loonman) [inline ShanghaiBloodSports/Assets/Scripts/MockInputBuffer.cs]: I really like the idea of mocking, but does unity support subfolders?
e.g. Scripts/InputBuffer/MockInputBuffer.cs

We're starting to get big enough that I think we should look into organizing the scripts a bit more

Comment (dktrotti) [reply ShanghaiBloodSports/Assets/Scripts/MockInputBuffer.cs]: I agree, let's do that in another pull request

File: ShanghaiBloodSports/Assets/Scripts/AutoRotator.cs (hunk 2 above)
Comment (dktrotti) [reply ShanghaiBloodSports/Assets/Scripts/AutoRotator.cs]: 😍

**Comment by Loonman** [comment]
Dom, movie time

**Pull request merged**